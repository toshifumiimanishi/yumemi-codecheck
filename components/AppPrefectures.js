const AppPrefectures = () => (
  <div className="prefecture">
    <div className="prefecture_container">
      <h2 className="prefecture_heading">都道府県</h2>
      <ul className="prefecture_list">
        <li className="prefecture_item">
          <label>
            <input type="checkbox" />
          </label>
        </li>
      </ul>
    </div>
  </div>
);

export default AppPrefectures;
