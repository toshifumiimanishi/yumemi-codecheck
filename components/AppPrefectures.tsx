import styled from '@emotion/styled';
import { Prefecture } from '../interfaces';

type ContainerProps = {
  prefectures: Prefecture[],
  fetchPopulationComposition: (prefecture: Prefecture) => void,
  removeTotalPopulation: (prefCode: number) => void
};

type Props = {
  className?: string,
  prefectures: Prefecture[],
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void
};

const AppPrefectures: React.FC<Props> = ({
  className,
  prefectures,
  handleChange
}) => (
  <section className={className}>
    <h2>都道府県</h2>
    <ul>
      {
        prefectures.map(prefecture => (
          <li key={prefecture.prefCode}>
            <label>
              <input type="checkbox" value={prefecture.prefCode} name={prefecture.prefName} onChange={handleChange} />
              {prefecture.prefName}
            </label>
          </li>
        ))
      }
    </ul>
  </section>
);

const StyledAppPrefectures = styled(AppPrefectures)`
  margin: auto;
  width: 100%;
  max-width: 960px;

  > h2 {
    display: inline-block;
    margin-bottom: 24px;
    padding: 0 8px;
    font-size: 24px;
    background-color: #000;
    color: #fff;
  }

  > ul {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(160px, auto));
    gap: 24px;
  }
`

const ContainerAppPrefectures: React.FC<ContainerProps> = ({
  prefectures,
  fetchPopulationComposition,
  removeTotalPopulation
}) => {
  const handleChange = (event) => {
    const prefName = event.currentTarget.name;
    const prefCode = event.currentTarget.value;
    const isChecked = event.currentTarget.checked;

    if (isChecked) {
      fetchPopulationComposition({ prefName, prefCode });
    } else {
      removeTotalPopulation(prefCode);
    }
  };

  return (
    <StyledAppPrefectures
      prefectures={prefectures}
      handleChange={handleChange}
    />
  )
};

export default ContainerAppPrefectures;
