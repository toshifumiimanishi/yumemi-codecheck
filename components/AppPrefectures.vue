<template>
  <div class="prefecture">
    <div class="prefecture_container">
      <h2 class="prefecture_heading">都道府県</h2>
      <ul class="prefecture_list">
        <li class="prefecture_item" v-for="prefecture in prefectures" :key="prefecture.prefCode">
          <label>
            <input type="checkbox" :value="prefecture.prefName" v-model="checkedPrefName">
            {{ prefecture.prefName }}
          </label>
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    prefectures: {
      type: Array
    }
  },
  data() {
    return {
      checkedPrefName: []
    }
  }
}
</script>

<style lang="scss">
.prefecture_container {
  margin: auto;
  width: 100%;
  max-width: 960px;
}

.prefecture_heading {
  display: inline-block;
  margin-bottom: 24px;
  padding: 0 8px;
  font-size: 24px;
  background-color: #000;
  color: #fff;
}

.prefecture_list {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(160px, auto));
  gap: 24px;
}
</style>
