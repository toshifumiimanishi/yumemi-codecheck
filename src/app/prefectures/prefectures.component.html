<div class="prefecture">
  <div class="prefecture_container">
    <h2 class="prefecture_heading">都道府県</h2>
    <ul class="prefecture_list">
      <li class="prefecture_item" *ngFor="let prefecture of prefectures">
        <label>
          <input type="checkbox">
          {{ prefecture.prefName }}
        </label>
      </li>
    </ul>
  </div>
</div>
