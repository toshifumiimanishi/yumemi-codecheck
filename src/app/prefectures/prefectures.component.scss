.prefecture_container {
  margin: auto;
  width: 100%;
  max-width: 960px;
}

.prefecture_heading {
  display: inline-block;
  margin-bottom: 24px;
  padding: 0 8px;
  font-size: 24px;
  background-color: #000;
  color: #fff;
}

.prefecture_list {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(160px, auto));
  gap: 24px;
}
