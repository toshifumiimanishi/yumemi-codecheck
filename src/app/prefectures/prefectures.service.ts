import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../environments/environment';

@Injectable()
export class PrefecturesService {
  constructor(private http: HttpClient) {}

  async fetchPrefectures() {
    const { result }: any = await this.http
      .get(
        `${environment.apiUrl}/prefectures`,
        {
          headers: new HttpHeaders({
            'X-API-KEY': environment.apiKey,
          }),
        }
      )
      .toPromise()

    return result;
  }
}
