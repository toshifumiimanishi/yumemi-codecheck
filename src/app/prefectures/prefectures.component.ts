import { Component, EventEmitter, Output, OnInit } from '@angular/core';

import { PrefecturesService } from './prefectures.service';

@Component({
  selector: 'prefecture',
  templateUrl: './prefectures.component.html',
  styleUrls: ['./prefectures.component.scss']
})
export class PrefecturesComponent implements OnInit {
  prefectures;

  @Output() checked = new EventEmitter

  constructor(private prefecturesService: PrefecturesService) {}

  async ngOnInit() {
    this.prefectures = await this.prefecturesService.fetchPrefectures();
  }

  onCheckedPrefName(event) {
    const prefName = event.currentTarget.name;
    const prefCode = event.currentTarget.value;
    const isChecked = event.currentTarget.checked;

    if (isChecked) {
      this.checked.emit({ prefName, prefCode });
    }
  }
}
