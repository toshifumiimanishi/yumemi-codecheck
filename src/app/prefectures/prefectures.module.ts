import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';

import { PrefecturesComponent } from './prefectures.component';
import { PrefecturesService } from './prefectures.service';

@NgModule({
  imports: [
    CommonModule,
    HttpClientModule
  ],
  declarations: [
    PrefecturesComponent
  ],
  providers: [
    PrefecturesService
  ],
  exports: [
    PrefecturesComponent
  ]
})
export class PrefecturesModule {}
