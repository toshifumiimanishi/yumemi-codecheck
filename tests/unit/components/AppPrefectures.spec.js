import { shallowMount } from '@vue/test-utils';
import AppPrefectures from '~/components/AppPrefectures.vue';
import prefectures from '../_mockData/prefectures.json';

describe('AppPrefectures コンポーネントのテスト', () => {
  const propsData = {
    prefectures
  };

  it('props を正しく受け取れるか', () => {
    const wrapper = shallowMount(AppPrefectures, { propsData });
    expect(wrapper.props()).toEqual(propsData);
  });

  it('レンダリング結果のスナップショットテスト', () => {
    const wrapper = shallowMount(AppPrefectures, { propsData });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
