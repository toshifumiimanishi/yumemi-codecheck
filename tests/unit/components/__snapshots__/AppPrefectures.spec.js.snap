// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPrefectures コンポーネントのテスト レンダリング結果のスナップショットテスト 1`] = `
<div
  class="prefecture"
>
  <div
    class="prefecture_container"
  >
    <h2
      class="prefecture_heading"
    >
      都道府県
    </h2>
     
    <ul
      class="prefecture_list"
    >
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="北海道"
            type="checkbox"
            value="1"
          />
          
          北海道
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="青森県"
            type="checkbox"
            value="2"
          />
          
          青森県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="岩手県"
            type="checkbox"
            value="3"
          />
          
          岩手県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="宮城県"
            type="checkbox"
            value="4"
          />
          
          宮城県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="秋田県"
            type="checkbox"
            value="5"
          />
          
          秋田県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="山形県"
            type="checkbox"
            value="6"
          />
          
          山形県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="福島県"
            type="checkbox"
            value="7"
          />
          
          福島県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="茨城県"
            type="checkbox"
            value="8"
          />
          
          茨城県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="栃木県"
            type="checkbox"
            value="9"
          />
          
          栃木県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="群馬県"
            type="checkbox"
            value="10"
          />
          
          群馬県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="埼玉県"
            type="checkbox"
            value="11"
          />
          
          埼玉県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="千葉県"
            type="checkbox"
            value="12"
          />
          
          千葉県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="東京都"
            type="checkbox"
            value="13"
          />
          
          東京都
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="神奈川県"
            type="checkbox"
            value="14"
          />
          
          神奈川県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="新潟県"
            type="checkbox"
            value="15"
          />
          
          新潟県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="富山県"
            type="checkbox"
            value="16"
          />
          
          富山県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="石川県"
            type="checkbox"
            value="17"
          />
          
          石川県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="福井県"
            type="checkbox"
            value="18"
          />
          
          福井県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="山梨県"
            type="checkbox"
            value="19"
          />
          
          山梨県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="長野県"
            type="checkbox"
            value="20"
          />
          
          長野県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="岐阜県"
            type="checkbox"
            value="21"
          />
          
          岐阜県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="静岡県"
            type="checkbox"
            value="22"
          />
          
          静岡県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="愛知県"
            type="checkbox"
            value="23"
          />
          
          愛知県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="三重県"
            type="checkbox"
            value="24"
          />
          
          三重県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="滋賀県"
            type="checkbox"
            value="25"
          />
          
          滋賀県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="京都府"
            type="checkbox"
            value="26"
          />
          
          京都府
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="大阪府"
            type="checkbox"
            value="27"
          />
          
          大阪府
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="兵庫県"
            type="checkbox"
            value="28"
          />
          
          兵庫県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="奈良県"
            type="checkbox"
            value="29"
          />
          
          奈良県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="和歌山県"
            type="checkbox"
            value="30"
          />
          
          和歌山県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="鳥取県"
            type="checkbox"
            value="31"
          />
          
          鳥取県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="島根県"
            type="checkbox"
            value="32"
          />
          
          島根県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="岡山県"
            type="checkbox"
            value="33"
          />
          
          岡山県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="広島県"
            type="checkbox"
            value="34"
          />
          
          広島県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="山口県"
            type="checkbox"
            value="35"
          />
          
          山口県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="徳島県"
            type="checkbox"
            value="36"
          />
          
          徳島県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="香川県"
            type="checkbox"
            value="37"
          />
          
          香川県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="愛媛県"
            type="checkbox"
            value="38"
          />
          
          愛媛県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="高知県"
            type="checkbox"
            value="39"
          />
          
          高知県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="福岡県"
            type="checkbox"
            value="40"
          />
          
          福岡県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="佐賀県"
            type="checkbox"
            value="41"
          />
          
          佐賀県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="長崎県"
            type="checkbox"
            value="42"
          />
          
          長崎県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="熊本県"
            type="checkbox"
            value="43"
          />
          
          熊本県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="大分県"
            type="checkbox"
            value="44"
          />
          
          大分県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="宮崎県"
            type="checkbox"
            value="45"
          />
          
          宮崎県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="鹿児島県"
            type="checkbox"
            value="46"
          />
          
          鹿児島県
        
        </label>
      </li>
      <li
        class="prefecture_item"
      >
        <label>
          <input
            name="沖縄県"
            type="checkbox"
            value="47"
          />
          
          沖縄県
        
        </label>
      </li>
    </ul>
  </div>
</div>
`;
